src/elements/content-picker/ContentPicker.js

```diff
@@ -845,13 +901,31 @@ class ContentPicker extends Component<Props, State> {
         }
 
         this.api.getAPI(type).share(item, access, (updatedItem: BoxItem) => {
-            this.refreshCollection();
+            this.updateItemInCollection(updatedItem);
             if (item.selected) {
-                this.select(updatedItem);
+                this.select(updatedItem, false);
             }
         });
     };
 
+    /**
+     * Updates the BoxItem in the state's currentCollection
+     *
+     * @param {Object} item file or folder object
+     * @returns {void}
+     */
+    updateItemInCollection = (item: BoxItem) => {
+        const { currentCollection } = this.state;
+        const { items = [] } = currentCollection;
```

Comment:
Can `items` be undefined here?

Reply:
It's unlikely that `items` could be undefined here, do you see a potential issue?